peps/pep-0757.rst

```diff
@@ -259,32 +257,21 @@ Code::
     mpz_set_PyLong(mpz_t z, PyObject *obj)
     {
         static PyLongExport long_export;
-
-        PyLong_Export(obj, &long_export);
-        if (long_export.digits) {
-            mpz_import(z, long_export.ndigits, int_digits_order, int_digit_size,
-                       int_endianness, int_nails, long_export.digits);
-            if (long_export.negative) {
+        const PyLongExport_Kind kind = PyLong_Export(obj, &long_export);
+
+        switch (kind) {
+        case PyLongExport_DigitArray:
+            mpz_import(z, long_export.digit_array.ndigits, int_digits_order,
+                       int_digit_size, int_endianness, int_nails,
+                       long_export.digit_array.digits);
+            if (long_export.digit_array.negative) {
                 mpz_neg(z, z);
             }
             PyLong_FreeExport(&long_export);
-        }
-        else {
-            const int64_t value = long_export.value;
-
-            if (LONG_MIN <= value && value <= LONG_MAX) {
-                mpz_set_si(z, value);
-            }
-            else {
-                mpz_import(z, 1, -1, sizeof(int64_t), 0, 0, &value);
-                if (value < 0) {
-                    mpz_t tmp;
-                    mpz_init(tmp);
-                    mpz_ui_pow_ui(tmp, 2, 64);
-                    mpz_sub(z, z, tmp);
-                    mpz_clear(tmp);
-                }
-            }
+            break;
+        default:
+            PyLong_FreeExport(&long_export);
+            abort();  /* new CPython release come with a new format */
```

Comment:
Sorry,
```suggestion
            abort();  /* new CPython release came with a new format */
```
?

The point is, PyLong_Export() - shouldn't fail on integers.  But it's permitted that we can introduce a new export format with the new release.  Users can put upper bound on supported CPython versions (then this branch will be unreachable) or just call ``abort()``.